stack/rush-stack-compiler/src/ApiExtractorRunner.ts

```diff
@@ -0,0 +1,231 @@
+   */
+  publishFolderForInternal?: string;
+
+  /**
+   * This setting is only used if dtsRollupTrimming is true.
+   * It indicates the folder where "npm publish" will be run for a beta release.
+   * The default value is "./dist/beta".
+   *
+   * @beta
+   * A beta release will contain all definitions that are reachable from the entry point,
+   * except definitions marked as \@alpha or \@internal.
+   */
+  publishFolderForBeta?: string;
+
+  /**
+   * This setting is only used if dtsRollupTrimming is true.
+   * It indicates the folder where "npm publish" will be run for a public release.
+   * The default value is "./dist/public".
+   *
+   * @beta
+   * A public release will contain all definitions that are reachable from the entry point,
+   * except definitions marked as \@beta, \@alpha, or \@internal.
+   */
+  publishFolderForPublic?: string;
+
+  /**
+   * This option causes the typechecker to be invoked with the --skipLibCheck option. This option is not
+   * recommended and may cause API Extractor to produce incomplete or incorrect declarations, but it
+   * may be required when dependencies contain declarations that are incompatible with the TypeScript engine
+   * that API Extractor uses for its analysis. If this option is used, it is strongly recommended that broken
+   * dependencies be fixed or upgraded.
+   */
+  skipLibCheck?: boolean;
+}
+
+/**
+ * The ApiExtractorTask uses the api-extractor tool to analyze a project for public APIs. api-extractor will detect
+ * common problems and generate a report of the exported public API. The task uses the entry point of a project to
+ * find the aliased exports of the project. An api-extractor.ts file is generated for the project in the temp folder.
+ * @beta
+ */
+export class ApiExtractorRunner extends RushStackCompilerBase<IApiExtractorTaskConfig> {
+  constructor(taskOptions: IApiExtractorTaskConfig, constants: Constants, terminalProvider: ITerminalProvider) {
+    super(taskOptions, constants, terminalProvider);
+  }
+
+  public invoke(): Promise<void> {
+    if (!this._validateConfiguration()) {
+      return Promise.resolve();
+    }
+
+    if (!this._taskOptions.entry) {
+      return Promise.reject(new Error('entry must be defined'));
+    }
+
+    if (!this._taskOptions.apiJsonFolder) {
+      return Promise.reject(new Error('apiJsonFolder must be defined'));
+    }
+
+    if (!this._taskOptions.apiReviewFolder) {
+      return Promise.reject(new Error('apiReviewFolder must be defined'));
+    }
+
+    try {
+      let entryPointFile: string;
+
+      if (this._taskOptions.entry === 'src/index.ts') {
+        // backwards compatibility for legacy projects that used *.ts files as their entry point
+        entryPointFile = path.join(this._constants.projectFolderPath, 'lib/index.d.ts');
+      } else {
+        entryPointFile = path.join(this._constants.projectFolderPath, this._taskOptions.entry);
+      }
+
+      const extractorConfig: IExtractorConfig = {
```

Comment:
>const extractorConfig: IExtractorConfig = { [](start = 6, length = 43)

Can we do this in the legacy code, and pass a conventional `IExtractorConfig` into the public API for RS?